text/1026-net-module.rst

```diff
@@ -0,0 +1,200 @@
+::
+
+    Status: Draft
+    Type: Feature
+    Created: 2024-08-14
+    Authors: Scott Trinh <scott@edgedb.com>
+
+=============================
+RFC: EdgeDB Networking Module
+=============================
+
+This RFC outlines the addition of a networking module to EdgeDB’s
+standard library, providing functionality for making HTTP requests and
+sending SMTP messages.
+
+Motivation
+==========
+
+The EdgeDB Auth extension (ext::auth) currently sends emails directly to
+configured SMTP services at certain times in various authentication
+ceremonies and workflows:
+
+-  Sign up with email (sends email verification email)
+-  Password reset (sends password reset email)
+-  Magic link sign up (sends magic link email)
+-  Magic link sign in (sends magic link email)
+
+To allow full customization of emails and provide hooks into
+authentication events generally, we need to support sending webhook
+requests to the application directly. This would enable applications to
+trigger their own logic for sending fully customized and
+internationalized emails, SMS, log analytics, update usage billing
+records, etc.
+
+Specification
+=============
+
+Module Name
+-----------
+
+The new module will be named ``std::net``.
+
+Functions
+---------
+
+The module will provide the following functions:
+
+1. HTTP Request Function:
+
+.. code-block:: edgeql
+
+   scalar type std::net::HttpMethod extending std::enums<GET, POST, PUT, DELETE, PATCH, HEAD, OPTIONS>;
```

Comment:
`GET` and `DELETE` are reserved words. Does this make it more awkward to use due to needing to quote it, or is it fine since `GET` is the default and `DELETE` is relatively less common to use?